src/instructlab/configuration.py

```diff
@@ -358,13 +410,26 @@ def init(ctx, config_file):
         log.configure_logging(log_level=config_obj.general.log_level.upper())
         ctx.obj = Lab(config_obj)
         # default_map holds a dictionary with default values for each command parameters
-        training_dict = get_dict(ctx.obj.config)
+        config_dict = get_dict(ctx.obj.config)
         # since torch and train args are sep, they need to be combined into a single `train` entity for the default map
         # this is because the flags for `ilab model train` will only be populated if the default map has a single `train` entry, not two.
-        training_dict["train"] = (
-            training_dict["train"]["train_args"]
-            | training_dict["train"]["torch_args"]
-            | training_dict["train"]["train_args"]["lora"]
-            | training_dict["train"]["train_args"]["deepspeed_options"]
+        config_dict["train"] = (
+            config_dict["train"]["train_args"]
+            | config_dict["train"]["torch_args"]
+            | config_dict["train"]["train_args"]["lora"]
+            | config_dict["train"]["train_args"]["deepspeed_options"]
         )
-        ctx.default_map = training_dict
+        config_dict["evaluate"] = (
+            config_dict["evaluate"]
+            | config_dict["evaluate"]["mmlu"]
+            | config_dict["evaluate"]["mmlu_branch"]
+            | config_dict["evaluate"]["mt_bench"]
+            | config_dict["evaluate"]["mt_bench_branch"]
+        )
+        # need to delete the individual sub-classes from the map
```

Comment:
since `_evaluate` has "top level" options that arent in sub classes, the `|` operator can't pull out primitive values (str, float, etc). So, what I do here is create a map with the entire old eval class, AND I add the flattened sub classes. so it looks something like 

` eval
    mt:
    mmlu
    mmlubranch
    mtbranch
    ALL_OPTIONS_FROM_THOSE_CLASSES_DUPLICATED_IN_TOP_LEVEL
`
I then delete mt, mmlu, and the branch benches so I am left with just the "unfolded" options in a "top level" eval key that can be directly mapped to the flags.

This is super confusing. I chatted with @nathan-weinberg about it, I can whiteboard it for folks on monday too

Reply:
That would be helpful, thank you :pray: